message/retrylater.go

```diff
@@ -0,0 +1,30 @@
+package message
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	servicebus "github.com/Azure/azure-service-bus-go"
+)
+
+// RetryLater waits for the given duration before retrying the processing of the message.
+// This happens in memory and does not impact servicebus message max retry limit
+func RetryLater(retryAfter time.Duration) Handler {
+	return &retryLaterHandler{
+		retryAfter: retryAfter,
+	}
+}
+
+type retryLaterHandler struct {
+	retryAfter time.Duration
+}
+
+func (r *retryLaterHandler) Do(ctx context.Context, orig Handler, message *servicebus.Message) Handler {
+	select {
+	case <-ctx.Done():
+		return Error(fmt.Errorf("aborting retrylater: %w", ctx.Err()))
```

Comment:
probably better off not wrapping, to make error handling simpler from the user perspective.